src/api/workshop.rs

```diff
@@ -141,6 +176,114 @@ pub mod workshop {
         }
     }
 
+    #[napi]
+    pub fn update_item_with_callback(
+        item_id: BigInt,
+        update_details: UgcUpdate,
+        app_id: Option<u32>,
+
+        #[napi(
+            ts_arg_type = "(data: { itemId: bigint; needsToAcceptAgreement: boolean }) => void"
+        )]
+        success_callback: napi::JsFunction,
```

Comment:
Do we really need to receive a JsFunction and transform it to threadsafe on these callbacks? Maybe it can be handled automatically by napi? https://napi.rs/docs/concepts/function

Reply:
Yes, Using a JsFunction and converting it to a thread-safe function is necessary, and this is related to the following code: 
https://github.com/Noxime/steamworks-rs/blob/master/src/ugc.rs
```rust
    pub fn submit<F>(self, change_note: Option<&str>, cb: F) -> UpdateWatchHandle<Manager>
    where
        F: FnOnce(Result<(PublishedFileId, bool), SteamError>) + 'static + Send,
    {
        use std::ptr;
        unsafe {
            let change_note = change_note.and_then(|v| CString::new(v).ok());
            let note = change_note.as_ref().map_or(ptr::null(), |v| v.as_ptr());
            let api_call = sys::SteamAPI_ISteamUGC_SubmitItemUpdate(self.ugc, self.handle, note);
            register_call_result::<sys::SubmitItemUpdateResult_t, _, _>(
                &self.inner,
                api_call,
                CALLBACK_BASE_ID + 4,
                move |v, io_error| {
                    cb(if io_error {
                        Err(SteamError::IOFailure)
                    } else if v.m_eResult != sys::EResult::k_EResultOK {
                        Err(v.m_eResult.into())
                    } else {
                        Ok((
                            PublishedFileId(v.m_nPublishedFileId),
                            v.m_bUserNeedsToAcceptWorkshopLegalAgreement,
                        ))
                    })
                },
            );
        }
        UpdateWatchHandle {
            ugc: self.ugc,
            _inner: self.inner,
            handle: self.handle,
        }
    }
```